### PR TITLE
feat(rdb): Adding privileges support

### PR DESCRIPTION
**Pull request by jeansebastienh**
## Resources
- [x] Implementation
- [x] Test
- [x] Documentation

## Datasources
- [x] Implementation
- [x] Test
- [x] Documentation

### CODE REVIEW

File: docs/data-sources/rdb_privilege.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- `user_name` - (Required) The user name.
+
+- `database_name` - (Required) The database name.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+- `permission` - The permission for this user on the database
```
Comment (remyleone) [inline docs/data-sources/rdb_privilege.md:35]: Is it permission or a privilege?

Comment (remyleone) [reply docs/data-sources/rdb_privilege.md:35]: Could you add valid values ? Valid values are `readonly`, `readwrite`, `all`, `custom` and `none`

File: scaleway/resource_rdb_privilege.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+	return nil
+}
+
+func resourceScalewayRdbPrivilegeDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	_ = d.Set("permission", rdb.PermissionNone)
+	return resourceScalewayRdbPrivilegeUpdate(ctx, d, meta)
+}
+
+func validationPrivilegePermission() func(interface{}, string) ([]string, []error) {
```
Comment (remyleone) [inline scaleway/resource_rdb_privilege.go:140]: Could you use `validation.StringInSlice` https://pkg.go.dev/github.com/hashicorp/terraform/helper/validation#StringInSlice ?

File: scaleway/resource_rdb_privilege_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+		if len(databases.Privileges) != 1 {
+			return fmt.Errorf("no privilege found")
+		}
+
+		return nil
+	}
+}
+
+func TestValidationPrivilegePermissionWithWrongPermissionReturnError(t *testing.T) {
```
Comment (remyleone) [inline scaleway/resource_rdb_privilege_test.go:147]: Why are those tests required if you use validation defined in the schema?

File: docs/data-sources/rdb_privilege.md (hunk 1 above)
Comment (jeansebastienh) [reply docs/data-sources/rdb_privilege.md:35]: As it's the data source: it's an exported value. Is it useful ?

File: scaleway/resource_rdb_privilege_test.go (hunk 3 above)
Comment (jeansebastienh) [reply scaleway/resource_rdb_privilege_test.go]: Unit testing :)

But following https://github.com/scaleway/terraform-provider-scaleway/pull/844/commits/046230f1cc4255ae7fdb23fe3e0d700fac63cee9 those tests are removed

File: docs/data-sources/rdb_privilege.md (hunk 1 above)
Comment (remyleone) [reply docs/data-sources/rdb_privilege.md:35]: it will be a different page on the website :) that's why it is useful to put it in both places

**Review by remyleone — APPROVED** [review]

**Comment by juliuste** [comment]
Hey 🙂 Thank you very much for implementing this! Any estimations when a new provider version which includes this feature will be published to the terraform registry? 😄

**Comment by remyleone** [comment]
The new provider is coming very soon, we are in the process of fixing some acceptance tests and will do a release soon after. If you are in a hurry and need to have this provider ASAP you can use the following override to make it happen: https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers

**Pull request merged**